lib/app/services/media_service/media_compress_service.dart

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:io';
+import 'dart:ui';
+
+import 'package:cross_file/cross_file.dart';
+import 'package:ffmpeg_helper/ffmpeg_helper.dart';
+import 'package:ffmpeg_wasm/ffmpeg_wasm.dart';
+import 'package:file_saver/file_saver.dart';
+import 'package:flutter/foundation.dart';
+import 'package:ice/app/services/logger/logger.dart';
+import 'package:ice/app/services/media_service/ffmpeg_args/ffmpeg_audio_bitrate_arg.dart';
+import 'package:ice/app/services/media_service/ffmpeg_args/ffmpeg_audio_codec_arg.dart';
+import 'package:ice/app/services/media_service/ffmpeg_args/ffmpeg_bitrate_arg.dart';
+import 'package:ice/app/services/media_service/ffmpeg_args/ffmpeg_preset_arg.dart';
+import 'package:ice/app/services/media_service/ffmpeg_args/ffmpeg_scale_arg.dart';
+import 'package:ice/app/services/media_service/ffmpeg_args/ffmpeg_video_codec_arg.dart';
+import 'package:image/image.dart' as img;
+import 'package:image_compression_flutter/image_compression_flutter.dart' as compressor_package;
+import 'package:path/path.dart' as path;
+import 'package:path_provider/path_provider.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'media_compress_service.g.dart';
+
+class MediaCompressionService {
+  Future<XFile> compressVideo(
+    XFile inputFile, {
+    FFmpegVideoCodecArg videoCodec = FFmpegVideoCodecArg.libx264,
+    FfmpegPresetArg preset = FfmpegPresetArg.fast,
+    FfmpegBitrateArg videoBitrate = FfmpegBitrateArg.medium,
+    FfmpegBitrateArg maxRate = FfmpegBitrateArg.medium,
+    FfmpegBitrateArg bufSize = FfmpegBitrateArg.medium,
+    FfmpegScaleArg scale = FfmpegScaleArg.fullHd,
+    int fps = 30,
+    FfmpegAudioCodecArg audioCodec = FfmpegAudioCodecArg.aac,
+    FfmpegAudioBitrateArg audioBitrate = FfmpegAudioBitrateArg.low,
+    bool overwrite = true,
+  }) async {
+    final args = [
+      const CustomArgument(['-threads', '0']),
+      if (overwrite) const CustomArgument(['-y']),
+      CustomArgument(['-codec:v', videoCodec.codec]),
+      CustomArgument(['-preset', preset.value]),
+      CustomArgument(['-b:v', videoBitrate.bitrate]),
+      CustomArgument(['-maxrate', maxRate.bitrate]),
+      CustomArgument(['-bufsize', bufSize.bitrate]),
+      CustomArgument(['-vf', 'scale=${scale.resolution},fps=$fps']),
+      CustomArgument(['-codec:a', audioCodec.codec]),
+      CustomArgument(['-b:a', audioBitrate.bitrate]),
+    ];
+
+    if (kIsWeb) {
+      return _compressVideoForWeb(inputFile, args);
+    }
+
+    if (Platform.isWindows) {
+      await _initWindowsPackage();
+    }
+
+    final output = await _generateOutputPath();
```

Comment:
Extract this block to some `_defaultVideoCompress` to be consistent with `_compressVideoForWeb`

Reply:
done 👍